pkg/cluster/self_hosted.go

```diff
@@ -43,24 +43,32 @@ func (c *Cluster) addOneSelfHostedMember() error {
 	}
 	// wait for the new pod to start and add itself into the etcd cluster.
 	oldN := c.members.Size()
-	err = retryutil.Retry(5*time.Second, math.MaxInt64, func() (bool, error) {
+	err = retryutil.Retry(10*time.Second, 6, func() (bool, error) {
 		err = c.updateMembers(c.members)
 		if err != nil {
-			c.logger.Errorf("add self hosted member: fail to update members: %v", err)
-			if err == errUnexpectedUnreadyMember {
-				return false, nil
-			}
-			return false, err
+			c.logger.Warningf("unable to update members: %v", err)
+			return false, nil
 		}
 		if c.members.Size() > oldN {
 			return true, nil
 		}
+		c.logger.Infof("current members: %v, need %d members, re-updating members later", c.members, oldN+1)
 		return false, nil
 	})
 	if err != nil {
-		return err
+		c.logger.Warningf("failed to add member (%s) due to scheduling failure. Removing its pod", newMember.Name)
```

Comment:
These two lines are different.
For this specific line, we did actually fail to wait the member to be added.
For line 69, we didn't remove the member, but only the pod.